provider/pkg/schema/gen.go

```diff
@@ -847,37 +849,26 @@ func (ctx *context) gatherResourceType() error {
 		}
 		addUntypedPropDocs(propertySpec, ctx.cfTypeName)
 		properties[sdkName] = *propertySpec
-		if createOnlyProperties.Has(sdkName) || !readOnlyProperties.Has(prop) {
+		if requiredProperties.Has(prop) || readOnlyProperties.Has(prop) {
+			required.Add(sdkName)
```

Comment:
Read-only properties are required? I don't understand that part.

Reply:
Yeah, I'm not 100% sure on the reason for this logic - that read-only properties are guarenteed to be present as outputs. This just maintains the current behaviour which was previously written as:
```go
	for prop := range readOnlyProperties {
		sdkName := ToSdkName(prop)
		if _, has := properties[sdkName]; has {
			required.Add(sdkName)
		}
	}
```